proposals/POEM-3-action-limit-for-namespace.md

```diff
@@ -0,0 +1,283 @@
+# limitations under the License.
+#
+-->
+
+# Title
+Providing action limits for each namespace
+
+## Status
+* Current state: In-progress
+* Author(s): @upgle
+
+## Summary and Motivation
+
+This POEM proposes a new feature that allows administrators to set action limits (memory, timeout, log, and concurrency) for each namespace.
+
+Sometimes some users want to make an action with more memory and longer duration. But, Openwhisk only has a system limit for action shared by all namespaces.
+There is no way to adjust the action limit for a few users, and changing the action limit setting will affect all users.
+
+In some private environments, you can operate Openwhisk more flexibly by providing different action limits.
+(For example, providing high memory only to some users.)
+
+```
+          256M                               512M
+
+            │     namespace default limit      │
+            ▼                                  ▼
+ ┌──────────┬──────────────────────────────────┬────────────┬────────────────────────┐
+ │          │┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼┼│----------► │                        │
+ └──────────┴──────────────────────────────────┴────────────┴────────────────────────┘
+ ▲                                                          ▲                        ▲
+ │ system limit                                             │ namespace limit        │ system limit
+
+128M                                                      1024M                    2048M
+```
+
+## Proposed changes
+
+### 3 types of action limits
+
+There was only a system limit shared by all namespaces, but two more concepts for namespace limits are added.
+
+- (1) system limit: It can never be exceeded under any circumstances.
+- (2) namespace default limit: It can be used if a limit has not been set for a namespace.
+- (3) namespace limit: It can be set by a system administrator for a namespace and cannot exceed the range of the system limit.
+
+### Limit configs for namespace
+
+- The maxParameterSize, maxPayloadSize and truncationSize values are treated as `ByteSize` string. (e.g. 1 MB, 512 KB...)
+
+The following settings are new:
+
+config key             | Type                         | description
+---------------------- | ---------------------------- | ---------------
+minActionMemory        | integer (unit: MB)           | minimum action memory size for namespace
+maxActionMemory        | integer (unit: MB)           | maximum action memory size for namespace
+minActionLogs          | integer (unit: MB)           | minimum activation log size for namespace
+maxActionLogs          | integer (unit: MB)           | maximum activation log size for namespace
+minActionTimeout       | integer (unit: milliseconds) | minimum action time limit for namespace
+maxActionTimeout       | integer (unit: milliseconds) | maximum action time limit for namespace
+minActionConcurrency   | integer                      | minimum action concurrency limit for namespace
```

Comment:
I think this should be renamed. This is representing container activation concurrency. If we are ever to add action concurrency like namespace concurrency but more fine grained, this will then be very confusing